stake-pool/program/src/processor.rs

```diff
@@ -2171,7 +2171,12 @@ impl Processor {
             return Err(StakePoolError::InvalidState.into());
         }
 
-        let pool_tokens_fee = if stake_pool.manager_fee_account == *burn_from_pool_info.key {
+        // To prevent a faulty manager fee account from preventing withdrawals
+        // if the token program does not own the account, or if the account is not initialized
+        let pool_tokens_fee = if stake_pool.manager_fee_account == *burn_from_pool_info.key
+            || check_account_owner(manager_fee_info, &stake_pool.token_program_id).is_err()
+            || Account::unpack(&manager_fee_info.data.borrow())?.state != AccountState::Initialized
```

Comment:
We should bundle these checks to be reused in other places, like deposit, sol deposit, and update stake pool balance.  The most important one is update stake pool balance, because failing that would also brick the pool for everyone.

`Account::unpack()` can fail for a lot of reasons, so we could still run into this issue with this change.  To make sure that `manager_fee_info` is valid, we need:

* owned by token_program_id
* unpacks to `Account`
* state is `Initialized`
* mint is correct

if any of those fails, give back `0`

Reply:
Ok, done